### PR TITLE
EVG-18246 Upgrade storybook to 7.0 beta

### PR DESCRIPTION
**Pull request by khelif96**
EVG-18246

### Description 
Upgrade storybook to the latest beta release candidate

- [X] Bump storybook to 7.0 beta
- [X] Migrate storybook config to ts
- [X] Ensure eslint runs on storybook file
- [X] Refactor stories to use new storybook format
- [X] Render storyshots using react-testing-library renderer to get a more accurate reflection of rendered html

### Screenshots
Figured since the 7.0 release is so soon it might be safe to upgrade to the 7.0 beta since it would make the official 7.0 upgrade easier when we do it in the future.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
   },
   typescript: {
     reactDocgen: "react-docgen-typescript",
     reactDocgenTypescriptOptions: {
       compilerOptions: {
         allowSyntheticDefaultImports: false,
-        esModuleInterop: false,
-      },
-    },
+        esModuleInterop: false
```
Comment (sophstad) [inline .storybook/main.js:23]: This is a nit/partially my own curiosity, but it seems like now we're applying different lint rules in this file than elsewhere in the repo (i.e. no trailing comma in an object like this). It'd be nice to have everything be standardized if you can figure out why our linter is being overriden.

File: src/components/Icon/Icon.stories.tsx [hunk 2]
```diff
@@ -16,28 +16,29 @@ const Sizes = {
 export default {
   title: "Components/Icon",
```
Comment (minnakt) [inline src/components/Icon/Icon.stories.tsx:17]: I noticed that some of the story files have titles, while others do not. Did you mean to delete the titles? (this applies to multiple files)

```suggestion

```

File: .storybook/main.js (hunk 1 above)
Comment (khelif96) [reply .storybook/main.js:23]: Good catch i'll take a look at the linter rules

File: src/components/Icon/Icon.stories.tsx [hunk 3]
```diff
@@ -16,28 +16,29 @@ const Sizes = {
...
-
-Default.argTypes = {
-  color: { control: "color" },
-  size: { control: { type: "select", options: Sizes } },
 };
-Default.args = {
-  color: green.dark3,
-  size: Sizes[Size.XLarge],
+
+export const Default: StoryObj<typeof Icon> = {
```
Comment (minnakt) [inline src/components/Icon/Icon.stories.tsx:21]: the new format is cool, especially the play functions 😳

File: .storybook/main.js [hunk 4]
```diff
@@ -1,46 +1,46 @@
...
-  framework: "@storybook/react",
-  core: {
-    builder: "@storybook/builder-vite",
+  addons: ["@storybook/addon-essentials", "@storybook/addon-interactions", "@storybook/addon-links",
+  // React 18 causes some issues with storyshots. In package.json we have a resolutions field for
+  // react-test-renderer to get around this. (https://github.com/storybookjs/storybook/issues/17985)
+  "@storybook/addon-storyshots"],
+  framework: {
+    name: "@storybook/react-vite",
+    options: {}
```
[the remark this answers is not in the document]
Comment (khelif96) [reply .storybook/main.js]: Nope the type definition requires it.

File: .storybook/main.js [hunk 5]
```diff
@@ -1,46 +1,46 @@
...
-    // react-test-renderer to get around this. (https://github.com/storybookjs/storybook/issues/17985)
-    "@storybook/addon-storyshots",
-  ],
-  framework: "@storybook/react",
-  core: {
-    builder: "@storybook/builder-vite",
+  addons: ["@storybook/addon-essentials", "@storybook/addon-interactions", "@storybook/addon-links",
+  // React 18 causes some issues with storyshots. In package.json we have a resolutions field for
+  // react-test-renderer to get around this. (https://github.com/storybookjs/storybook/issues/17985)
+  "@storybook/addon-storyshots"],
```
[the remark this answers is not in the document]
Comment (khelif96) [reply .storybook/main.js]: Good catch

File: .storybook/main.js (hunk 1 above)
Comment (khelif96) [reply .storybook/main.js]: https://stackoverflow.com/questions/57947585/eslint-warning-file-ignored-by-default-use-a-negated-ignore-pattern

Looks like eslint ignores files that start with `.` by default and you have to intentionally negate the ignored file.

File: src/storyshots.test.ts [hunk 6]
```diff
@@ -1,19 +1,26 @@
...
+      const { container } = render(component);
+      // wait until the mount is updated
+      const waitTime = 1;
+      setTimeout(() => {
+        if (snapshotFileName) {
+          expect(container).toMatchSpecificSnapshot(snapshotFileName);
+        }
+
+        done?.();
+      }, waitTime);
```
Comment (minnakt) [inline src/storyshots.test.ts:23]: Can you expand/elaborate on this comment for this block of code?

**Review by minnakt — APPROVED** [review]

**Review by sophstad — APPROVED** [review]

**Pull request merged**